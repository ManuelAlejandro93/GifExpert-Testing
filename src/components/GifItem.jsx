import PropTypes from 'prop-types';

const GifItem = ({ url, title }) => {
	return (
		<div className='card'>
			<img
				src={url}
				alt={title}
			/>
			<p>{title}</p>
		</div>
	);
};

export default GifItem;

GifItem.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};
