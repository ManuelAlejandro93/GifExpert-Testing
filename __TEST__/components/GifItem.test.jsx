import { render, screen } from '@testing-library/react';

import GifItem from '../../src/components/GifItem';

describe('Tests on </GifItem>', () => {
	//Arrange.
	const mockUrl = 'https://www.google.com/';
	const mockTitle = 'estoy en google';

	test('el titulo debe existir en la pantalla.', () => {
		//Act.
		render(
			<GifItem
				url={mockUrl}
				title={mockTitle}
			/>,
		);

		//Assert.
		expect(screen.queryByText(/estoy en google/i)).toBeTruthy();
	});

	test('data should be an object js element', () => {
		//Act.
		render(
			<GifItem
				url={mockUrl}
				title={mockTitle}
			/>,
		);

		const { src, alt } = screen.getByRole('img');
		//Assert.
		expect(src).toBe(mockUrl);
		expect(alt).toBe(mockTitle);
	});

	test('Props must be required.', () => {
		//Act.
		const { container } = render(<GifItem />);
		//Assert.
		expect(screen.queryByRole('img').innerHTML === '').toBeTruthy();
	});

	// InstantÃ¡nea del componente.
	test('el componente debe coinncidir con su foto.', () => {
		const { container: syntaxComponent } = render(
			<GifItem
				url={mockUrl}
				title={mockTitle}
			/>,
		);

		expect(syntaxComponent).toMatchSnapshot();
	});
});
