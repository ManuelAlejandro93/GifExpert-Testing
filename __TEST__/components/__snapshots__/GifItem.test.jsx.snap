// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on </GifItem> el componente debe coinncidir con su foto. 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="estoy en google"
      src="https://www.google.com/"
    />
    <p>
      estoy en google
    </p>
  </div>
</div>
`;
